WalletWasabi/Blockchain/Transactions/TransactionStore.cs

```diff
@@ -49,10 +51,16 @@ public TransactionStore(string workFolderPath, Network network, bool migrateData
 	/// <remarks>Guarded by <see cref="SqliteStorageLock"/>.</remarks>
 	private Dictionary<uint256, SmartTransaction> Transactions { get; } = new();
 
-	private void Import(string oldPath, Network network, bool deleteAfterImport = false)
+	// ToDo: Temporary to fix https://github.com/zkSNACKs/WalletWasabi/pull/12137#issuecomment-1879798750
+	public bool NeedResync { get; private set; }
+
+	private void Import(string oldPath, string dbPath, Network network, bool deleteAfterImport = true)
 	{
 		if (File.Exists(oldPath))
 		{
+			// ToDo: Temporary to fix https://github.com/zkSNACKs/WalletWasabi/pull/12137#issuecomment-1879798750
+			NeedResync = File.Exists(dbPath);
```

Comment:
If the file exists, it means that either
1) A problem occurred during migration -> this would be considered as a false positive
2) The client ran the toxic version, `deleteAfterImport`was `false`, and needs resync -> this is what we want to catch